Lib/venv/__init__.py

```diff
@@ -41,11 +41,13 @@ class EnvBuilder:
                      environment
     :param prompt: Alternative terminal prefix for the environment.
     :param upgrade_deps: Update the base venv modules to the latest on PyPI
+    :param gitignore: Create a .gitignore file in the environment directory
+                      which causes it to be ignored by git.
     """
 
     def __init__(self, system_site_packages=False, clear=False,
                  symlinks=False, upgrade=False, with_pip=False, prompt=None,
-                 upgrade_deps=False):
+                 upgrade_deps=False, *, gitignore=False):
```

Comment:
I'm going to be controversial here, but shouldn't we add type annotations for new code? 😆

Reply:
The stdlib is type annotated in typeshed -- see e.g. https://discuss.python.org/t/21487

Reply:
That thread seems more of a discussion starter than decision. Also admints in there that parts of the stdlib already do have type annotations 🤔 I'd like at some point to be able to stop needing typeshed. typeshed always plays catch-up and gets out of sync with the code, so IMHO doing it in the code is better.

Reply:
Eh, I don't think my comment was off-topic. Is very much on topic for this PR, that is why I belive we should add type annotations here. I'd prefer if you'd offer the option to me to resolve conversations, when I feel we discussed enough the topic, rather than you deciding... This feels very much like dismissing my comment without allowing it to be discussed just because you don't agree with my stance.

Reply:
Some modules in the stdlib (`importlib.metadata`, `asyncio`) have partial type annotations, and one (`tomllib`) has complete type-annotation coverage. However, no annotations added to the stdlib are checked by any type checker, and in fact all annotations added inline to stdlib functions are completely ignored by type checkers when they check usages of these functions in third-party code. When it comes to the stdlib, type checkers only look at typeshed's annotations.

As such, while type hints in the stdlib aren't completely banned, they're not particularly useful in most cases, and so the current policy is that they aren't generally encouraged. This is especially when adding type annotations contradicts the style of the surrounding code, which it does here. (Most of the modules that have heavy use of type annotations, such as `importlib.metadata` and `tomllib`, have third-party backports that are kept in sync with the stdlib code — the type annotations are checked in the third-party repos.)

If you'd like to debate a change in that policy and argue that we should always add type annotations in new code, we can have that discussion, but it shouldn't be done here, on this PR thread. You'd need to open a Discourse thread and try to get consensus among the core devs before we could change this policy. Until the policy is changed, Adam is correct that adding a type annotation wouldn't be appropriate here. Debate over the general policy CPython currently has, and over the merits and demerits of typeshed, is unfortunately off-topic for a review of this PR.

Reply:
It's codified based on discussions among the core developers that we have all agreed to over the years, but writing it down wouldn't hurt.

Reply:
> I think your comment should be part of https://devguide.python.org so it's more authorative 🤔 then a still open discussion.

Formalising our policy is a reasonable ask, but adding something to the devguide will still require getting consensus among the core devs... I'll try to muster the energy for that, but no promises :)